bin/process-images.js

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env node
+/**
+ * process-images.js
+ * Normalises source image filenames.
+ */
+(function (_, fs, fse) {
+    'use strict';
+    var processImages;
+
+    // Generate "products.csv" based on the source data.
+    processImages = function (sourcePath, targetPath) {
+        console.log('Processing images with sourcePath="%s", targetPath="%s"', sourcePath, targetPath);
+        fs.readdir(sourcePath, function (err, sourceImages) {
+            var copyNextImage;
+            if (err) {
+                console.error('Could not read images path "%s", please check the path exists and try again', sourcePath);
+                return;
+            }
+            copyNextImage = function () {
+                var sourceImage;
+                if (sourceImages.length === 0) {
+                    return;
+                }
+                sourceImage = sourceImages.pop();
+                fse.copySync(sourcePath + '/' + sourceImage, targetPath + '/' + sourceImage.replace(/[^-\w.\/]+/g, '_'));
```

Comment:
Nah.  First empirically:

    'foo bar\nbaz ^caret .dot'.replace(/[^-\w.\/]+/g, '_');
    'foo_bar_baz_caret_.dot'

(The dot is allowed through as we want).

Outside of a character group, `.` means "match anything" which is actually a shorthand for a character group that includes every possible character.  Inside a character group, it only matches a `.` character.  In general, you only need to escape things in character groups that would otherwise have another meaning.  In the case of `.`, it never means "match anything" inside a character group, because "match anything" is itself a character group (and if you think about it, it's pretty meaningless to use "match anything" inside another character group because anything else you add will already be covered, which then makes the whole group construct also redundant).

To take a weirder example, the `^` character needs to be escaped or not depending on its _position_ in the group: `/[a^]/` and `/\^a]/` are equivalent (match `a` or `^`), but `/[^a]/` is completely different (match anything except `a`):

    > 'foo bar\nbaz ^caret .dot'.replace(/[\^a]/g, '_');
    'foo b_r\nb_z _c_ret .dot'
    
    > 'foo bar\nbaz ^caret .dot'.replace(/[a\^]/g, '_');
    'foo b_r\nb_z _c_ret .dot'
    
    > 'foo bar\nbaz ^caret .dot'.replace(/[^a]/g, '_');
    '_____a___a____a________'

So, in summary, regular expressions are quite irregular at times.